dali/base/dadfs.cpp

```diff
@@ -12854,6 +12855,28 @@ IPropertyTreeIterator *deserializeFileAttrIterator(MemoryBuffer& mb, unsigned nu
                 file->setPropBool(getDFUQResultFieldName(DFUQRFiscompressed), true);
         }
 
+        void setCost(IPropertyTree* file, const char *nodeGroup)
+        {
+            StringBuffer str;
+            double fileAgeDays = 0.0;
+            if (file->getProp(getDFUQResultFieldName(DFUQRFtimemodified), str))
+            {
+                CDateTime dt;
+                dt.setString(str.str());
+                fileAgeDays = difftime(time(nullptr), dt.getSimple())/(24*60*60);
+            }
+            __int64 sizeDiskSize = 0;
+            if (isCompressed(*file))
+                sizeDiskSize = file->getPropInt64(getDFUQResultFieldName(DFUQRFcompressedsize), 0);
+            else
+                sizeDiskSize = file->getPropInt64(getDFUQResultFieldName(DFUQRForigsize), 0);
+            double sizeGB = sizeDiskSize / ((double)1024 * 1024 * 1024);
+            __int64 numDiskWrites = file->getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskReads), 0);
+            __int64 numDiskReads = file->getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskWrites), 0);
+            double totalCost = calcFileCost(nodeGroup, sizeGB, fileAgeDays, numDiskWrites, numDiskReads);
+            file->addPropReal(getDFUQResultFieldName(DFUQRFcost), totalCost);
```

Comment:
trivial: this should really be setPropReal , addProp* implies adding multiple property values of the specified name.